internal/ingress/controller/template/template.go

```diff
@@ -296,16 +296,17 @@ func needsRewrite(location *ingress.Location) bool {
 	return false
 }
 
-// atLeastOneNeedsRewrite checks if the nginx.ingress.kubernetes.io/rewrite-target annotation is used on the '/' path
-func atLeastOneNeedsRewrite(input interface{}) bool {
+// atLeastOneUsesRegex checks if the "rewrite-target" or "use-regex" annotation
+// is used on any location path within a server
+func atLeastOneUsesRegex(input interface{}) bool {
```

Comment:
nitpick: but something like `enforceRegexModifier` describes what this function does better